Feat/enterprise callout

ThisDavidRichard opened the pull request:
Summary
  - Adds a reusable enterprise callout component to promote enterprise 
  features across documentation
  - Supports 4 variants: migration support, AI Agent beta, Pages beta, and 
  Semantic Search beta
  - Enables manual placement control by content editors

  ## Implementation Details

  ### Components Added
  - **EnterpriseCallout**: Base component with variant-specific content and
   styling
  - **EnterpriseCalloutAuto**: Wrapper component that handles placement 
  (inline vs sidebar)

  ### Features
  - **Explicit control**: Content editors specify both variant and 
  placement via props
  - **Portal rendering**: Sidebar callouts render in the actual right 
  sidebar using React Portals
  - **Two placement modes**:
    - `inline`: Renders at the bottom of page content (for pages without 
  sidebars)
    - `sidebar`: Renders in the right sidebar next to table of contents
  - **Consistent design**: Thin border, rounded corners, primary CTA, and 
  waitlist option

  ### Pages Updated
  - **Migration guides**: All 9 migration guides now have sidebar callouts
  - **AI Agent**: Overview (inline) and Install (sidebar) pages
  - **Pages**: Overview (inline) and Install (sidebar) pages  
  - **Semantic Search**: Documentation page (sidebar)

  ### Usage
  \`\`\`tsx
  // For sidebar placement
  <EnterpriseCalloutAuto variant=\"migration\" placement=\"sidebar\" />

  // For inline placement
  <EnterpriseCalloutAuto variant=\"ai-agent\" placement=\"inline\" />
  \`\`\`

  ## Testing
  - Verified callouts render correctly in both sidebar and inline positions
  - Tested across all updated documentation pages
  - Confirmed portal rendering works for sidebar placement"

ThisDavidRichard committed:
Add enterprise callout component for migration, AI Agent, Pages, and Semantic Search features

- Create EnterpriseCallout component with variant-specific content
- Add EnterpriseCalloutAuto wrapper that auto-detects variant based on page URL
- Support both sidebar and inline rendering modes
- Integrate callouts across migration guides, AI Agent, Pages, and Semantic Search docs
- Include primary CTA for enterprise customers and waitlist option for other plans

ThisDavidRichard committed:
Add enterprise callout component for migration, AI Agent, Pages, and Semantic Search features

- Create EnterpriseCallout component with variant-specific content
- Add EnterpriseCalloutAuto wrapper that auto-detects variant based on page URL
- Placement (sidebar vs inline) is explicitly controlled via inline prop
- Integrate callouts in Layout sidebar for pages with right sidebar
- Add inline callouts to pages without sidebar (AI Agent overview, Pages overview)
- Include primary CTA for enterprise customers and waitlist option for other plans

ThisDavidRichard committed:
Merge branch 'feat/enterprise-callout' of https://github.com/ueberdosis/tiptap-docs into feat/enterprise-callout

ThisDavidRichard committed:
Add enterprise callout component system with manual placement control

  - Create EnterpriseCallout base component with 4 variants (migration,
  ai-agent, pages, semantic-search)
  - Add EnterpriseCalloutAuto wrapper with explicit variant and placement
  props (inline/sidebar)
  - Use React Portal to render sidebar callouts in the right sidebar DOM
  location
  - Add inline callouts to pages without sidebars (AI Agent overview, Pages
   overview)
  - Add sidebar callouts to all migration guides, AI Agent install, Pages
  install, and Semantic Search pages
  - Remove auto-detection logic - editors have full control via props

ThisDavidRichard committed:
linting

bdbch commented on src/components/EnterpriseCallout.tsx:
```diff
@@ -0,0 +1,136 @@
+            ? 'Document layouts'
+            : 'AI document search'
+
+    const description =
```
Do we want those descriptions to be hard-coded / fixed or should the be passed in as a prop?

bdbch commented on src/components/EnterpriseCallout.tsx:
```diff
@@ -0,0 +1,136 @@
+            ? 'Get real-time integration support in a shared Slack channel for Enterprise customers.'
+            : 'Semantic Search is currently available to Enterprise customers as part of a limited rollout.'
+
+    const waitlistText =
```
Same here? Maybe we can keep those variants but also allow to pass in a custom text + description

bdbch commented on src/components/EnterpriseCalloutAuto.tsx:
```diff
@@ -0,0 +1,52 @@
+
+export type EnterpriseCalloutAutoProps = {
+  variant: 'migration' | 'ai-agent' | 'pages' | 'semantic-search'
+  placement?: 'inline' | 'sidebar'
```
Is this hard related to the layout or can this be used in the content? Maybe we can rename this to indicate more what it will do instead of where it will be placed?

bdbch reviewed (changes requested):
Just nitpicks

ThisDavidRichard committed:
make EnterpriseCallout props configurable and rename placement to renderMode

  - Add optional title, description, and waitlistText props to
  EnterpriseCallout component
  - Rename placement prop to renderMode in EnterpriseCalloutAuto for better
   semantic clarity
  - Update all MDX files to use new renderMode prop instead of placement

ThisDavidRichard committed:
Lint

ThisDavidRichard committed:
add disableWaitlist prop to EnterpriseCallout and apply to docx pages

  - Add disableWaitlist prop to conditionally hide waitlist CTA
  - Update EnterpriseCalloutAuto to support docx variant and pass through
  prop
  - Rename renderMode from 'portal-sidebar' to 'sidebar' for clarity
  - Disable waitlist on all docx documentation pages
